### ISSUE TITLE
Fetch FW emulator builds from GitHub ci insted of Gitlab 

### ISSUE THREAD

**Issue by vdovhanych** [trezor/trezor-user-env#233]
As GitLab will be completely deprecated in our setup and will happen in a couple of months, we have to change where main builds are getting fetched from.

**Comment by grdddj** [comment]
Makes sense ... I already added T3T1 build into Gitlab, but then I realized I missed the ARM version ... probably it does not make sense to add it there anymore, when we can already implement the GH-Actions emulators

@mmilata Are there some static links in GH-Actions for the latest main run, same as in Gitlab?

**Comment by vdovhanych** [comment]
We will have to use github api for that so i think we need to create a custom API call to get the latest artifacts from a build. Or use something like [this](https://github.com/dawidd6/action-download-artifact) to download the artifacts during build action and integrate them in the docker container.

**Comment by grdddj** [comment]
> We will have to use github api for that so i think we need to create a custom API call to get the latest artifacts from a build. Or use something like [this](https://github.com/dawidd6/action-download-artifact) to download the artifacts during build action and integrate them in the docker container.

Right, I am looking into the API approach --- got it almost working, but the final archive downloading does not work without the `GH_TOKEN` : https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28#download-an-artifact

We will probably need to generate one with read access specific for this purpose --- and then leaving it in CI so it loads it from there when building. I would also need it for testing purposes (my own token did not work)

**Comment by mmilata** [comment]
We can add a workflow that will push the emulators to data.trezor.io whenever `main` is updated.

**Comment by mmilata** [comment]
If we end up using artifacts it would probably be a good idea to increase `retention-days` from the current 7.

**Comment by vdovhanych** [comment]
I think it should be fine as we should fetch latest artifacts and this runs every night so we would be getting new artifacts every night.

**Comment by grdddj** [comment]
> We can add a workflow (or rather a job to core.yml) that will push the emulators to data.trezor.io whenever `main` is updated.

That would be awesome, it should be easier to push the emulators when we have them present, rather than trying to fetch them dynamically by searching the latest jobs. 

Also, `tenv` would like to get `legacy` latest build ... that might be questionable, given the small amount of changes it receives.


>  But lest try first using artifacts and see how that goes

I have requested a token with read access to `firmware` GH-Actions, please approve it @vdovhanych

**Comment by grdddj** [comment]
Thanks for the token, it work well, at least locally.

Changes outlined in PR - https://github.com/trezor/trezor-user-env/pull/234

Can we please set the `GH_TOKEN` env variable in GH `secrets`, so we can use it within the download script?

**Comment by grdddj** [comment]
Fetching partly from `trezor.io` and partly still from `GitLab` - done in https://github.com/trezor/trezor-user-env/pull/234
